src/ray/common/network_util.cc

```diff
@@ -71,3 +72,13 @@ bool Ping(const std::string &ip, int port, int64_t timeout_ms) {
   bool is_timeout;
   return client.Connect(ip, port, timeout_ms, &is_timeout);
 }
+
+bool CheckFree(int port) {
```

Comment:
Hmm, this might introduce a new race condition - the OS sometimes doesn't mark ports as free until after a delay, so it could be that the worker fails to bind because we are binding here. Should be able to address this by adding a short retry/timeout in the worker.

Reply:
Addressed belwo